sky/data/storage.py

```diff
@@ -1616,9 +1616,25 @@ def get_dir_sync_command(src_dir_path, dest_dir_name):
             # we exclude .git directory from the sync
             excluded_list = storage_utils.get_excluded_files(src_dir_path)
             excluded_list.append('.git/*')
+
+            # Process exclusion patterns to make them work correctly with aws
+            # s3 sync
+            processed_excludes = []
+            for excluded_path in excluded_list:
+                # Check if the path is a directory exclusion pattern
+                # For AWS S3 sync, directory patterns need to end with "/**" to
+                # exclude all contents
+                if (excluded_path.endswith('/') or os.path.isdir(
+                        os.path.join(src_dir_path, excluded_path.rstrip('/')))):
+                    # Remove any trailing slash and add '/**' to exclude all
+                    # contents
+                    processed_excludes.append(f'{excluded_path.rstrip("/")}/**')
```

Comment:
not a blocker - do we need `/**` or is `/*` sufficient? see https://github.com/skypilot-org/skypilot/blob/bdc880a5ff3b8f7d89a7f788a351f9f31b04a493/sky/data/storage_utils.py#L213-L216

Reply:
Good catch! This part should be general. Just submitted a commit for this. Also, tested that single '*' works.

Reply:
Actually, adding patterns could cause the handling of our zip upload to fail as we don't use pattern matching in the underlying code. I am removing the pattern matching globally. Running the smoke tests for storage again.